drivers/docker/network.go

```diff
@@ -29,35 +29,52 @@ func (d *Driver) CreateNetwork(allocID string) (*drivers.NetworkIsolationSpec, e
 	}
 	_, err = d.coordinator.PullImage(d.config.InfraImage, authOptions, allocID, noopLogEventFn)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	config, err := d.createSandboxContainerConfig(allocID)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
-	container, err := d.createContainer(client, *config, d.config.InfraImage)
+	specFromContainer := func(c *docker.Container) *drivers.NetworkIsolationSpec {
```

Comment:
You could probably just create `container` here and skip the func, but this is a neat way to skip memory allocations for structs that may not be used.

Reply:
I was a little wary of creating a partial `NetworkIsolationSpec` and then populating the fields later just because it's the sort of thing that's easy for someone to come along later and return it early and expose a nil pointer exception in a field access. If you think the indirection of the function here isn't idiomatic I'd be happy to change it though.